File: elixir/bob/bob.exs
defmodule Teenager do

  def hey(input) do
    cond do
      silent?(input) -> "Fine. Be that way!"
      question?(input) -> "Sure."
      shouting?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

  defp silent?(input) do
    "" == String.strip(input)
  end

  defp question?(input) do
    String.ends_with?(input, "?")
  end

  defp shouting?(input)  do
    Regex.match?(~r/\p{L}+/, input) && String.equivalent?(String.upcase(input), input)
  end

end

File: clojure/bob/bob.clj
(ns bob)

  (defn question? [speech]
    (boolean (re-find #"(\?$)" speech))
  )

  (defn yelling? [speech]
    (and (= speech (clojure.string/upper-case speech)) (boolean (re-find #"[a-zA-Z]" speech)))
  )

  (defn response-for [speech]
    (cond
      (clojure.string/blank? speech) "Fine. Be that way!"
      (and (question? speech) (not (yelling? speech))) "Sure."
      (yelling? speech) "Whoa, chill out!"
      :else "Whatever.")
  )

